src/policy_artifacthub.rs

```diff
@@ -0,0 +1,325 @@
+                        }
+                        false => artifacthubpkg.links = None,
+                    }
+                }
+                policy_metadata::Annotation::Keywords(s) => {
+                    let mut rdr = ReaderBuilder::new()
+                        .flexible(true)
+                        .from_reader(s.as_bytes());
+
+                    let mut keywords: Vec<String> = vec![];
+                    for row in rdr.deserialize() {
+                        keywords.push(row.expect("annotation \"io.artifacthub.keywords\" in policy metadata is malformed, must be csv values"));
+                    }
+                    artifacthubpkg.keywords = Some(keywords);
+                }
+                policy_metadata::Annotation::PolicyUsage(s) => {
+                    if artifacthubpkg.readme.is_none() {
+                        artifacthubpkg.readme = Some(s.to_string())
+                    }
+                }
+                policy_metadata::Annotation::OCIURL(u) => {
+                    let oci_url = Url::parse(format!("{}:v{}", u, version).as_str()).expect("annotation \"io.kubewarden.policy.ociUrl\" in policy metadata must be a well formed url");
+                    let container_images = vec![ContainerImage {
+                        name: "policy".to_string(),
+                        image: oci_url.to_string(),
+                    }];
+                    artifacthubpkg.containers_images = Some(container_images);
+                }
+                policy_metadata::Annotation::Resources(s) => {
+                    _ = artifacthubpkg
+                        .annotations
+                        .insert(Annotation::KubewardenResources(s.to_string()))
+                }
+                policy_metadata::Annotation::HiddenUI(b) => {
+                    _ = artifacthubpkg
+                        .annotations
+                        .insert(Annotation::RancherHiddenUi(*b));
+                }
+            }
+        }
+
+        // check all required fields are correctly set:
+        if artifacthubpkg.name.is_empty() {
+            return Err(anyhow!(
+                "policy metadata must specify {} in annotations",
+                "\"io.kubewarden.policy.title\"",
+            ));
+        }
+        if artifacthubpkg.display_name.is_empty() {
+            return Err(anyhow!(
+                "policy metadata must specify {} in annotations",
+                "\"io.artifacthub.displayName\"",
+            ));
+        }
+        if artifacthubpkg.description.is_empty() {
+            return Err(anyhow!(
+                "policy metadata must specify {} in annotations",
+                "\"io.kubewarden.policy.description\"",
+            ));
+        }
+        if artifacthubpkg.links.is_none() {
+            return Err(anyhow!(
+                "policy metadata must specify {} in annotations",
+                "\"io.kubewarden.policy.source\"",
+            ));
+        }
```

Comment:
This is ok, but we could maybe use [`validator`](https://crates.io/crates/validator) for that. This is a crate we already use inside of policy-evaluator

Reply:
I've made the checks in separate functions, by pattern matching instead of validate functions. Please tell me what you think about it.